suite-native/transactions/src/components/TransactionDetail/TransactionDetailInputsSheet.tsx

```diff
@@ -0,0 +1,112 @@
+
+type TransactionDetailInputsSheetProps = {
+    isVisible: boolean;
+    transaction: WalletAccountTransaction;
+    onSheetVisibilityChange: () => void;
+};
+
+type TransactionAddressAmountProps = {
+    address: string;
+    amount: string;
+};
+
+const addressAmountColumnStyle = prepareNativeStyle(_ => ({
+    maxWidth: '42.5%',
+}));
+
+const TransactionAddressAmount = ({ address, amount }: TransactionAddressAmountProps) => (
+    <Box>
+        <Text variant="hint" color="gray1000" numberOfLines={1} ellipsizeMode="middle">
+            {address}
+        </Text>
+        <Text variant="label" color="gray600">
+            {amount}
+        </Text>
+    </Box>
+);
+
+export const TransactionDetailInputsSheet = ({
+    isVisible,
+    onSheetVisibilityChange,
+    transaction,
+}: TransactionDetailInputsSheetProps) => {
+    const { applyStyle } = useNativeStyles();
+    const { CryptoAmountFormatter } = useFormatters();
+
+    const txInputs = transaction.details.vin;
+    const txOutputs = transaction.details.vin;
+
+    const formatCryptoInputAmount = (amount?: string) =>
+        CryptoAmountFormatter.format(formatNetworkAmount(amount ?? '0', transaction.symbol), {
+            symbol: transaction.symbol,
+        });
+    return (
+        <TransactionDetailSheet
+            isVisible={isVisible}
+            onVisibilityChange={onSheetVisibilityChange}
+            title="Inpust & outputs"
+            iconName="swap"
+            transactionId={transaction.txid}
+        >
+            <Box
+                flexDirection="row"
+                justifyContent="space-between"
+                marginHorizontal="small"
+                marginBottom="medium"
+            >
+                <Box flex={1} flexDirection="row" alignItems="center" paddingLeft="small">
+                    <Text variant="hint" color="gray600">
+                        Inputs · {txInputs.length}
+                    </Text>
+                    <Box marginLeft="small">
+                        <Icon name="receiveAlt" color="gray600" size="medium" />
+                    </Box>
+                </Box>
+
+                <Box flex={1} flexDirection="row" alignItems="center" paddingLeft="large">
+                    <Text variant="hint" color="gray600">
+                        Outputs · {txOutputs.length}
+                    </Text>
+                    <Box marginLeft="small">
+                        <Icon name="sendAlt" color="gray600" size="medium" />
+                    </Box>
+                </Box>
+            </Box>
+            <VStack marginBottom="medium">
+                <Card>
+                    <Box flexDirection="row" justifyContent="space-between">
+                        <Box style={applyStyle(addressAmountColumnStyle)}>
+                            {txInputs.map((input, index) => (
+                                <TransactionAddressAmount
+                                    key={input.hex || index}
```

Comment:
Please be very carefull with `index` as key and avoid it as much as possible. You should have available plenty of other options here, for example `input.txid` should work too.

Reply:
fixed in [360bdb3](https://github.com/trezor/trezor-suite/pull/7370/commits/360bdb31f3189bc481292921ffd57db6f4609ee1)